File: Test projects/.NET48/WinformsTestAppNet48Vb/Forms/TestForm.vb
Imports WinformsTestAppNet48Vb.WinformsTestAppNet48Vb.Services

Public Class TestForm

    Private ReadOnly _exampleService As IExampleService

    Public Sub New(exampleService As IExampleService)
        Me._exampleService = exampleService

        InitializeComponent()

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        _exampleService.DoSomething()

    End Sub

End Class

File: Test projects/.NET48/WinformsTestAppNet48CSharp/Forms/TestForm.cs
using System;
using System.Windows.Forms;
using WinformsTestApp.Services;

namespace WinformsTestApp.Forms
{
    public partial class TestForm : Form
    {
        private readonly IExampleService _exampleService;

        public TestForm(IExampleService exampleService)
        {
            _exampleService = exampleService;
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            _exampleService.DoSomething();
        }
    }
}
